### PR TITLE
Added shard statistics REST APIs

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Docs PR: https://github.com/arangodb/docs/pull/618

* Added new REST APIs for retrieving the sharding distribution:

  - GET `/_api/database/shardDistribution` will return the number of collections, shards, leaders and followers for the database it is run inside. The request can optionally be restricted to include data from only a single DB server, by passing the `DBserver` URL parameter.

    This API can only be used on coordinators.

  - GET `/_admin/cluster/shardDistribution` will return global statistics on the current shard distribution, showing the total number of databases, collections, shards, leaders and followers for the entire cluster. The results can optionally be restricted to include data from only a single DB server, by passing the `DBserver` URL parameter. By setting the `details` URL parameter, the response will not contain aggregates, but instead one entry per available database will be returned.

    This API can only be used in the `_system` database of coordinators, and requires admin user privileges.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/618

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (shell_client)

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13884/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13889/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13891/

**Comment by jsteemann** [comment]
Tests blue

File: tests/js/client/shell/shell-global-shard-statistics-cluster.js
Comment (hkernbach) [inline tests/js/client/shell/shell-global-shard-statistics-cluster.js:113]: You could put the expected logic to the above loop in Line: 95, but that is micro-optimization. Your call.

**Review by hkernbach — APPROVED** [review]
LGMT, just a very minor thingy. Fell free to ignore or adjust.

File: tests/js/client/shell/shell-global-shard-statistics-cluster.js
Comment (jsteemann) [inline tests/js/client/shell/shell-global-shard-statistics-cluster.js:94]:
```suggestion
        let expected = ["_system", cn];
        db._useDatabase("_system");
```

File: tests/js/client/shell/shell-global-shard-statistics-cluster.js
Comment (jsteemann) [inline tests/js/client/shell/shell-global-shard-statistics-cluster.js:97]:
```suggestion
          expected.push(cn + "test" + i);
        }
```

File: tests/js/client/shell/shell-global-shard-statistics-cluster.js
Comment (jsteemann) [inline tests/js/client/shell/shell-global-shard-statistics-cluster.js:114]:
```suggestion
```

File: tests/js/client/shell/shell-global-shard-statistics-cluster.js
Comment (jsteemann) [reply tests/js/client/shell/shell-global-shard-statistics-cluster.js]: Ok, I did that!

**Comment by jsteemann** [comment]
Rerunning Jenkins after large merge:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13898/

**Pull request merged**
